src/test/java/org/jabref/logic/journals/JournalAbbreviationRepositoryTest.java

```diff
@@ -193,14 +194,62 @@ void testJournalUnabbreviate() {
     void testJournalAbbreviateWithoutEscapedAmpersand() {
         BibDatabase bibDatabase = new BibDatabase();
         JournalAbbreviationRepository journalAbbreviationRepository = JournalAbbreviationLoader.loadBuiltInRepository();
-        UndoableAbbreviator undoableAbbreviator = new UndoableAbbreviator(journalAbbreviationRepository, AbbreviationType.DEFAULT);
+        UndoableAbbreviator undoableAbbreviator = new UndoableAbbreviator(journalAbbreviationRepository, AbbreviationType.DEFAULT, false);
 
-        BibEntry entryWithoutEscapedAmpersandInJournal = new BibEntry(StandardEntryType.Article);
-        entryWithoutEscapedAmpersandInJournal.setField(StandardField.JOURNAL, "ACS Applied Materials & Interfaces");
+        BibEntry entryWithoutEscapedAmpersandInJournal = new BibEntry(StandardEntryType.Article)
+            .withField(StandardField.JOURNAL, "ACS Applied Materials & Interfaces");
 
         undoableAbbreviator.abbreviate(bibDatabase, entryWithoutEscapedAmpersandInJournal, StandardField.JOURNAL, new CompoundEdit());
         BibEntry expectedAbbreviatedJournalEntry = new BibEntry(StandardEntryType.Article)
                 .withField(StandardField.JOURNAL, "ACS Appl. Mater. Interfaces");
         assertEquals(expectedAbbreviatedJournalEntry, entryWithoutEscapedAmpersandInJournal);
     }
+
+    @Test
+    void testJournalAbbreviateWithEmptyFJournal() {
+        BibDatabase bibDatabase = new BibDatabase();
+        JournalAbbreviationRepository journalAbbreviationRepository = JournalAbbreviationLoader.loadBuiltInRepository();
+        UndoableAbbreviator undoableAbbreviator = new UndoableAbbreviator(journalAbbreviationRepository, AbbreviationType.DEFAULT, true);
+
+        BibEntry entryWithoutEscapedAmpersandInJournal = new BibEntry(StandardEntryType.Article)
+            .withField(StandardField.JOURNAL, "ACS Applied Materials & Interfaces")
+            .withField(AMSField.FJOURNAL, "&nbsp;");
+
+        undoableAbbreviator.abbreviate(bibDatabase, entryWithoutEscapedAmpersandInJournal, StandardField.JOURNAL, new CompoundEdit());
```

Comment:
This looks like some really bad software design, to provide the CompoundEdit to some package, that should afaik reside in the logic package (abbreviator), but is located in the gui package. But it's not your fault, you are just testing a gui component. Yet it feels totally wrong

Reply:
Yeah, major refactoring needed ^^.

Reply:
Maybe, this could be the start to think about a new undo/redo concept.

Reply:
I agree with you, and this undo redo stuff is the only reason. We also have this FieldChange stuff that is kind of similar, but this seems to be not coupled to the undo redo stuff